srml/support/src/dispatch.rs

```diff
@@ -67,38 +67,122 @@ pub trait Parameter: Codec + Clone + Eq {}
-/// used for side effects.
+/// If the `origin` param is omitted, the macro adds it as the first parameter and adds `ensure_root(origin)`
+/// as the first line of the function.
 ///
-/// For every module an associated enum (usually "Call") is generated with every variant
-/// corresponding to a function of the module. This enum implements Callable and thus its values
-/// can be used as an extrinsic's payload.
+/// ```
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+///
+/// 		// Privileged function: the macro will add the `origin` param
+/// 		// and call `ensure_root(origin)`.
+///			fn my_function() -> Result {
+///				// Your implementation
+/// 			Ok(())
+/// 		}
+///		}
+/// }
+/// # fn main() {}
+/// ```
+///
+/// ## Multiple Module Instances Example
 ///
-/// The `on_initialize` and `on_finalize` functions are special, since it can either take no
-/// parameters, or one parameter, which has the runtime's block number type.
+/// A Substrate module can be built such that multiple instances of the same module can be used within a single
+/// runtime. For example, the [Balances module](../srml_balances/index.html) can be added multiple times to your
+/// runtime in order to support multiple, independent currencies for your blockchain. Here is an example of how
+/// you would declare such a module using the `decl_module!` macro:
 ///
-/// ### Module with instances
+/// ```
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, ensure_signed};
+/// # pub struct DefaultInstance;
+/// # pub trait Instance {}
+/// # impl Instance for DefaultInstance {}
+/// pub trait Trait<I: Instance=DefaultInstance>: system::Trait {}
 ///
-/// decl_module! support modules with instances with the following syntax: (DefaultInstance type is
-/// optionnal)
-/// ```nocompile
-/// pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {}
+/// decl_module! {
+/// 	pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {
+/// 		// Your implementation
+/// 	}
+/// }
+/// # fn main() {}
 /// ```
+///
+/// ## Reserved Functions
+///
+/// The following are reserved function signatures:
+///
+/// * `deposit_event`: Helper function for depositing an [event](https://docs.substrate.dev/docs/event-enum).
+/// If set to default, it will call `deposit_event` from the [System module](../srml_system/index.html).
+/// However, you can write your own implementation for events in your runtime.
+///
+/// The following reserved functions also take the block number (with type `T::BlockNumber`) as an optional input:
+///
+/// * `on_initialize`: Executes at the beginning of a block.
```

Comment:
All these three functions implement the corresponding traits, maybe we should write this as an explanation.